test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -154,14 +156,29 @@ public void testStreamContents(InputStreamProvider inputStreamProvider,
      */
     @Test
     public void testMoreThanTwoGB() throws IOException {
-        Path sourceFile = Files.createTempFile(CWD, "test2GBSource", null);
+        Path sourceFile = CWD.resolve(Path.of(String.format("test3GBSource%s.tmp",
+            String.valueOf(RND.nextInt(Integer.MAX_VALUE)))));
         try {
             // preparing two temporary files which will be compared at the end of the test
             Path targetFile = Files.createTempFile(CWD, "test2GBtarget", null);
             try {
-                // writing 3 GB of random bytes into source file
-                for (int i = 0; i < NUM_WRITES; i++)
-                    Files.write(sourceFile, createRandomBytes(BYTES_PER_WRITE, 0), StandardOpenOption.APPEND);
+                // calculate initial position to be just short of 2GB
+                final long initPos = 2047*BYTES_PER_WRITE;
+
+                // create the file with a hint to be sparse
+                try (FileChannel fc = FileChannel.open(sourceFile, CREATE_NEW,
+                    SPARSE, WRITE, APPEND);) {
+                    // set initial position to avoid writing nearly 2GB
+                    fc.position(initPos);
+
+                    // fill the remainder of the file with random bytes
+                    int nw = (int)(NUM_WRITES - initPos/BYTES_PER_WRITE);
+                    for (int i = 0; i < nw; i++) {
+                        ByteBuffer src =
+                            ByteBuffer.wrap(createRandomBytes(BYTES_PER_WRITE, 0));
```

Comment:
Probably not worth splitting this line either.

Reply:
I missed this line split; fixed.